### ISSUE TITLE
Fix S927 FN: Parameter name does not match name in interface

### ISSUE THREAD

**Issue by drieseng** [SonarSource/sonar-dotnet#9662]
### Description

**S927** is not reported when a parameter of a method that implements an interface does not match the name of the parameter on the interface.

### Repro steps

Compile the following code snippet:

```c#
public interface ITransformer<in T>
{
    void Transform(T input);
}

public sealed class TrainTransformer : ITransformer<string>
{
    public void Transform(string input2)
    {
    }
}
```

### Expected behavior

**S927** should be reported as the parameter name of the **Transform** method on **TrainTransformer** does not match the name on on the **ITransformer<T>** interface.

### Actual behavior

**S927** is not reported.

### Related information

* Visual Studio version: 17.11.2
* SonarAnalyzer.CSharp version: 9.32.0.97167
* Operating System: Windows 11

**Comment by gregory-paidis-sonarsource** [comment]
This seems to have to do with the generic `T`, as we fail to match the `T` in `T input` with the `string` in `string input2`.

**Comment by alex-meseldzija-sonarsource** [comment]
Hi @drieseng ,
This is [actually expected behavior](https://github.com/SonarSource/sonar-dotnet/issues/3047.)

The intention is that when an interface has generic types, the class implementing it can be more specific allowing easier readability.

We will update the RSPEC to clarify this.

Thanks for raising this :)

**Comment by alex-meseldzija-sonarsource** [comment]
https://github.com/SonarSource/rspec/pull/4274
